### PR TITLE
Added pipeline example (cleaned)

### PR DESCRIPTION
**Pull request by Mo-Fatah**
```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

```
This example links to the second point in #377  (Pipeline example). And a cleaner version of #400 

The example consists of three programs:

1. `generate-set`: generate a set of numbers from 2 to upper_bound, which is passed as an argument.
2. `keep-primes`: applies [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) algorithm to keep only prime numbers.

### CODE REVIEW

File: sdk/rust-examples/prime-numbers-generator/keep-primes/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   
+    // remove last comma, add end of line character
+    content.push('\n');
+
+    fs::write(FILENAME, content)?;
+    
+    Ok(())
+}
+
+// implements sieve_of_eratosthenes algorithm, and retain only prime numbers from nums
```
Comment (ShaleXIONG) [inline sdk/rust-examples/prime-numbers-generator/keep-primes/src/main.rs:50]: triple `///` for rustdoc

**Review by ShaleXIONG — DISMISSED** [review]

File: sdk/rust-examples/prime-numbers-generator/generate-set/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+use anyhow;
+
+const OUTPUT_FILENAME: &'static str = "/output/number-set.txt";
+
+fn main() -> anyhow::Result<()> {
+    let mut set: Vec<u32> = Vec::new();
+
+    let args: Vec<String> = std::env::args().collect();
+
+    let upper_limit = args.get(0).unwrap_or(&String::from("100")).to_owned().parse::<u32>()?;
```
Comment (dominic-mulligan-arm) [inline sdk/rust-examples/prime-numbers-generator/generate-set/src/main.rs:26]: Did you run `cargo fmt` on this file?

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
@ShaleXIONG are you happy with these changes?

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
+1 reached, merging.

**Pull request merged**